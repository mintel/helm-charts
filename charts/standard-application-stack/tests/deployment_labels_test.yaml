suite: Test Deployment Labels
release:
  namespace: test-namespace
tests:
  - it: Check default label behavior
    template: deployment.yaml
    set:
      global.name: app
      global.owner: sre
      global.application: application-name
      global.component: component-name
      global.clusterEnv: qa
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: metadata.name
          value: app
      - equal:
          path: metadata.labels["app.kubernetes.io/name"]
          value: app
      - equal:
          path: metadata.labels["app.mintel.com/owner"]
          value: sre
      - equal:
          path: metadata.labels["app.mintel.com/application"]
          value: application-name
      - equal:
          path: metadata.labels["app.mintel.com/component"]
          value: component-name
  - it: Check empty application and component labels both default to name
    template: deployment.yaml
    set:
      global.name: app
      global.owner: sre
      global.clusterEnv: qa
    asserts:
      - matchSnapshot: {} # Check for regressions and unexpected changes.
      - equal:
          path: metadata.name
          value: app
      - equal:
          path: metadata.labels["app.kubernetes.io/name"]
          value: app
      - equal:
          path: metadata.labels["app.mintel.com/application"]
          value: app
      - equal:
          path: metadata.labels["app.mintel.com/application"]
          value: app
