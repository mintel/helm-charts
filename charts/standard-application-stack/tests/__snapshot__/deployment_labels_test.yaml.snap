Check default label behavior:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app
        app.mintel.com/application: application-name
        app.mintel.com/component: component-name
        app.mintel.com/env: qa
        app.mintel.com/owner: sre
        app.mintel.com/region: ${CLUSTER_REGION}
        name: app
      name: app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: app
            app.mintel.com/application: application-name
            app.mintel.com/component: component-name
            app.mintel.com/env: qa
            app.mintel.com/owner: sre
            app.mintel.com/region: ${CLUSTER_REGION}
            name: app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: app-app
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
Check empty application and component labels both default to name:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        app.mintel.com/placeholder: placeholder
        secret.reloader.stakater.com/reload: app-app
      labels:
        app.kubernetes.io/component: app
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app
        app.mintel.com/application: app
        app.mintel.com/component: app
        app.mintel.com/env: qa
        app.mintel.com/owner: sre
        app.mintel.com/region: ${CLUSTER_REGION}
        name: app
      name: app
      namespace: test-namespace
    spec:
      minReadySeconds: 10
      replicas: 2
      selector:
        matchLabels:
          app.kubernetes.io/component: app
          app.kubernetes.io/name: app
      strategy:
        rollingUpdate:
          maxSurge: 15%
          maxUnavailable: 10%
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: app
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: app
            app.mintel.com/application: app
            app.mintel.com/component: app
            app.mintel.com/env: qa
            app.mintel.com/owner: sre
            app.mintel.com/region: ${CLUSTER_REGION}
            name: app
        spec:
          containers:
            - command:
                - /app/docker-entrypoint.sh
              env:
                - name: KUBERNETES_POD_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.namespace
                - name: APP_ENVIRONMENT
                  value: qa
                - name: RUNTIME_ENVIRONMENT
                  value: kubernetes
              envFrom:
                - secretRef:
                    name: app-app
              image: registry.gitlab.com/test:v0.0.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              name: main
              ports:
                - containerPort: 8000
                  name: http
              readinessProbe:
                httpGet:
                  path: /readiness
                  port: http
                  scheme: HTTP
                periodSeconds: 10
                timeoutSeconds: 1
              resources:
                limits: {}
                requests: {}
              securityContext:
                allowPrivilegeEscalation: false
                capabilities:
                  drop:
                    - ALL
                runAsNonRoot: true
                runAsUser: 1000
              startupProbe:
                failureThreshold: 60
                httpGet:
                  path: /healthz
                  port: http
                  scheme: HTTP
                periodSeconds: 5
          securityContext:
            runAsNonRoot: true
            runAsUser: 1000
          serviceAccountName: app
          terminationGracePeriodSeconds: 30
          topologySpreadConstraints:
            - labelSelector:
                matchLabels:
                  app.kubernetes.io/component: app
                  app.kubernetes.io/name: app
              maxSkew: 1
              topologyKey: topology.kubernetes.io/zone
              whenUnsatisfiable: DoNotSchedule
